gcovr/coverage.py

```diff
@@ -213,14 +278,23 @@ def branch_coverage(self):
 
     def decision_coverage(self):
         # type: () -> Tuple[int, int, int, Optional[float]]
-        total = len(self.decisions)
+        total = 0
         cover = 0
-        unchecked = 0
-        for decision in self.decisions.values():
-            if decision.is_covered:
-                cover += 1
-            elif decision.is_uncheckable:
-                unchecked += 1
+        unchecked = False
+        if self.decision is not None:
+            if self.decision.is_uncheckable:
+                unchecked = True
+            elif self.decision.is_conditional:
+                total = 2
+                if self.decision.count_true > 0:
+                    cover += 1
+                if self.decision.count_false > 0:
+                    cover += 1
+            elif self.decision.is_switch:
+                if self.decision.count > 0:
+                    cover += 1
```

Comment:
I think a switch-condition must set `total = 1`. The changes to the test reference files look suspicious.

Reply:
`+= 1` results in a 1 but we can also set it hard to 1.
In the test only unchec kable decisions are changed because ther was a true and false part and now it's only one decision, shall a uncheckable be counted twice?

Reply:
Forget my comment, the snippet is incrementing `cover`. You're right.